apis/vmware/v1beta1/vspheremachine_types.go

```diff
@@ -64,6 +65,34 @@ type VSphereMachineSpec struct {
 	// Volumes is the set of PVCs to be created and attached to the VSphereMachine
 	// +optional
 	Volumes []VSphereMachineVolume `json:"volumes,omitempty"`
+
+	// PowerOffMode describes the desired behavior when powering off a VM.
+	//
+	// There are three, supported power off modes: hard, soft, and
+	// trySoft. The first mode, hard, is the equivalent of a physical
+	// system's power cord being ripped from the wall. The soft mode
+	// requires the VM's guest to have VM Tools installed and attempts to
+	// gracefully shut down the VM. Its variant, trySoft, first attempts
+	// a graceful shutdown, and if that fails or the VM is not in a powered off
+	// state after reaching the GuestShutdownTimeout, the VM is halted.
+	//
+	// If omitted, the mode defaults to trySoft.
+	//
+	// +optional
+	// +kubebuilder:default=trySoft
+	PowerOffMode vmopv1alpha1.VirtualMachinePowerOpMode `json:"powerOffMode,omitempty"`
+
+	// GuestShutdownTimeout sets the wait timeout for guest shutdown. The
+	// VM will be powered off forcibly after the timeout if the VM is still
+	// up and running when the PowerOffMode is set to trySoft.
+	//
+	// This parameter only applies when the PowerOffMode is set to trySoft.
+	// The reconciler will wait infinitely when it is set to zero.
+	//
+	// If omitted, the timeout defaults to 5 minutes.
+	//
+	// +optional
+	GuestShutdownTimeout *metav1.Duration `json:"guestShutdownTimeout,omitempty"`
```

Comment:
As long as this is not used: let's drop it! cc @randomvariable

Reply:
Removed.